NewRelic/AutoInteractor.php

```diff
@@ -0,0 +1,148 @@
+<?php
+
+/*
+ * This file is part of Ekino New Relic bundle.
+ *
+ * (c) Ekino - Thomas Rabaix <thomas.rabaix@ekino.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Ekino\Bundle\NewRelicBundle\NewRelic;
+
+/**
+ * This interactor does never assume that the NewRelic extension is installed. It will check
+ * for the existence of each method EVERY time. This is a good interactor to use when you want
+ * to enable and disable the NewRelic extension without rebuilding your container.
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class AutoInteractor implements NewRelicInteractorInterface
+{
+    /**
+     * @var NewRelicInteractorInterface
+     */
+    private $interactor;
+
+    /**
+     * @param NewRelicInteractorInterface $interactor
+     */
+    public function __construct(NewRelicInteractorInterface $real, NewRelicInteractorInterface $fake)
+    {
+        $this->interactor = extension_loaded('newrelic') ? $real : $fake;
```

Comment:
What is the use case of this service ?

From one side the container builder uses "real" or "balckhole" whether the extension is loaded or not. and from the other side, we have this class which seams to do the same thing at runtime

Reply:
I know that this is super weird. But checking #127 and possible #82 this make sense. 
The use case is when you have your deployed application with the container built, and then for some reason you unload the NewRelic extension (strange, yes). 

If you have this scenario, then you want to postpone the decision of using blackhole or not to runtime instead of build time.

Reply:
Ok, I got your point . In summary:
- Auto is a "runtime" selector
- ContainerBuilder is a "buildtime" selector

Do we need to keep the containerBuilder couldn't we just rely on this new Service?

Reply:
Using the ContainerBuilder is the proper "symfony way" of doing it. It is also a (very) little bit faster since there is two fewer classes to load. 

I think we should promote ContainerBuilder only and then if you got this strange scenario then the AutoInteractor is a "duct tape" solution.